### PR TITLE
feat: performance

### PR DESCRIPTION
**Pull request by shahargl**
close https://github.com/keephq/keep/issues/1632

### CODE REVIEW

File: docker/entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+gunicorn "keep.api.api:get_app(mode='rest')" \
+  --bind 0.0.0.0:8080 \
+  --workers 4 \
+  -k uvicorn.workers.UvicornWorker \
+  -c /venv/lib/python3.11/site-packages/keep/api/config.py &
+
+# Check if REDIS is true, and start the worker mode if it is
+# I'm using random port 1234 because it doesn't matter
+if [ "$REDIS" = "true" ]; then
+  echo "Starting worker mode..."
```
Comment (talboren) [inline docker/entrypoint.sh:14]: Dunno why this change but small comment: arq is a cli tool and can be executed independently. Guincorn is redundant in this case ❤️

Comment (shahargl) [reply docker/entrypoint.sh:14]: Yea I know I didn't to keep it a minimal change.

The full context is wide by this specific change is because if there are lot of alerts in the queue, the HTTP requests are throttled and not served, which I wanted to change by separating between workers and REST processes.

**Review by talboren — APPROVED** [review]
LGTM

**Review by talboren — CHANGES_REQUESTED** [review]
1 thing before lgtming: looks like there's a migration needed here, no?

File: keep/api/api.py [hunk 2]
```diff
@@ -159,84 +161,100 @@ def get_app(
...
-                "role": user.role,
-            }
+                }
+
+    # WORKER MODE
+    elif mode == "worker":
+        # Initialize only the worker-related components
+        @app.on_event("startup")
+        async def start_worker():
+            if REDIS and WORKER_ENABLED:
```
Comment (Matvey-Kuk) [inline keep/api/api.py:252]: Looks like this PR is making WORKER_ENABLED env var obsolete, it used to serve the same purpose:
- REDIS & WORKER_ENABLED for worker
- REDIS & !WORKER_ENABLED for api only

File: docker/entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+echo "Starting REST mode..."
+gunicorn "keep.api.api:get_app(mode='rest')" \
+  --bind 0.0.0.0:8080 \
+  --workers 4 \
+  -k uvicorn.workers.UvicornWorker \
+  -c /venv/lib/python3.11/site-packages/keep/api/config.py &
+
+# Check if REDIS is true, and start the worker mode if it is
+# I'm using random port 1234 because it doesn't matter
+if [ "$REDIS" = "true" ]; then
```
Comment (Matvey-Kuk) [inline docker/entrypoint.sh:13]: I don't think REDIS could be definitive for worker, shouldn't api be able to schedule tasks for worker via redis?

Comment (Matvey-Kuk) [reply docker/entrypoint.sh:13]: Other words, API should be able to access REDIS
